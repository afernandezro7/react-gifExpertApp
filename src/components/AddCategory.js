import React, { useState } from 'react';
import PropTypes from 'prop-types';



export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('Introduce un Elemento')

    const handleInputChange= (e)=>{
        setinputValue(e.target.value);
    }

    const handleKeyEnter= (e)=>{
        e.preventDefault();

        if( inputValue.trim().length > 2){
            
            setCategories(cats=>[inputValue,...cats])
            setinputValue('Introduce un Elemento')
            document.getElementById('Myform').reset()
        }
    }

    return (
        
        <form id="Myform" onSubmit= { handleKeyEnter }>
            <input
                type="text"               
                placeholder={inputValue}
                
                onChange={ handleInputChange }
            />
        </form>
        
    )
}

AddCategory.propTypes= {
    setCategories: PropTypes.func.isRequired
}

