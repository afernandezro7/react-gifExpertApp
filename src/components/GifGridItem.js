import React from 'react'

export const GifGridItem = ({id,title,url}) => {
    return (    
        <div className="card animate__animated animate__rollIn animate__delay-2s">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
           
        </div>  
    )
}
