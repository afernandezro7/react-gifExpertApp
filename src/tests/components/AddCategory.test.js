import React from 'react';
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';

describe('Preba sobre AddCategory.js', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories= { setCategories }/>);

    beforeEach( ()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories= { setCategories }/>);
    } )




    test('Debe funcionar el componente', () => {    

        expect( wrapper ).toMatchSnapshot();
    });



    test('Debe cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');

        const value = 'Hola Mundo'
        input.simulate('Change', { target: {value} })


    })



    test('No debe postear la informaciÃ³n con submit', () => {
        
        wrapper.find('form').simulate('submit', {preventDefault: ()=>{}})

        expect( setCategories ).not.toHaveBeenCalled()
    })
    


    test('debe llamar setCategories y limpiar caja de texto', () => {
         
        const value = 'Hola Mundo';

        // 1. simular el inputChange
        wrapper.find('input').simulate('Change', { target: {value} })
        
        // 2. simular el submit
        wrapper.find('form').simulate('submit', {preventDefault: ()=>{}})
        
        // 3. setCategories debe estar llamado
        expect ( setCategories).toHaveBeenCalled();
        expect ( setCategories).toHaveBeenCalledTimes(1);
        expect ( setCategories).toHaveBeenCalledWith( expect.any(Function) );

        // 4. el valor del input debe estar ''
        expect( wrapper.find('input').prop('value')).toBe('')
    })
    
    
})
