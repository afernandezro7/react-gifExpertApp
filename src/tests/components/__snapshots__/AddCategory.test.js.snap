// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preba sobre AddCategory.js Debe funcionar el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
